modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/agg/AccumulatorsFactory.java

```diff
@@ -229,14 +229,11 @@ private Function<Object[], Object[]> createInAdapter(Accumulator accumulator) {
             List<Function<Object, Object>> casts =
                     Commons.transform(Pair.zip(inTypes, outTypes), AccumulatorsFactory::cast);
 
-            return new Function<Object[], Object[]>() {
-                @Override
-                public Object[] apply(Object[] args) {
-                    for (int i = 0; i < args.length; i++) {
-                        args[i] = casts.get(i).apply(args[i]);
-                    }
-                    return args;
+            return args -> {
```

Comment:
I'm not sure what needs to be changed, anonymous classes can be faster than lambdas because of the lambda cache.